callouts/go/extproc/examples/client/client.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2024 Google LLC.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"io"
+	"log"
+	"net"
+	"sync"
+	"time"
+
+	extproc "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+// server is an implementation of extproc.ExternalProcessorServer.
+type server struct {
+	extproc.UnimplementedExternalProcessorServer
+}
+
+// Process processes incoming gRPC streams.
+func (s *server) Process(stream extproc.ExternalProcessor_ProcessServer) error {
+	for {
+		req, err := stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			return err
+		}
+
+		log.Printf("Received request: %v", req)
+
+		// Respond with a dummy ProcessingResponse for demonstration
+		response := &extproc.ProcessingResponse{
+			Response: &extproc.ProcessingResponse_RequestHeaders{
+				RequestHeaders: &extproc.HeadersResponse{},
+			},
+		}
+		if err := stream.Send(response); err != nil {
+			return err
+		}
+	}
+}
+
+// startServer starts a gRPC server at the specified address.
+func startServer(wg *sync.WaitGroup, address string) {
+	lis, err := net.Listen("tcp", address)
+	if err != nil {
+		log.Fatalf("Failed to listen: %v", err)
+	}
+	grpcServer := grpc.NewServer()
+	extproc.RegisterExternalProcessorServer(grpcServer, &server{})
+	log.Printf("Server listening at %v", lis.Addr())
+	wg.Done()
+	if err := grpcServer.Serve(lis); err != nil {
+		log.Fatalf("Failed to serve gRPC: %v", err)
+	}
+}
+
+var (
+	tls        = flag.Bool("tls", false, "Connection uses TLS if true, else plain TCP")
+	certFile   = flag.String("cert_file", "", "The file containing the CA root cert file")
+	serverAddr = flag.String("addr", "localhost:8181", "The server address in the format of host:port")
+	dataJSON   = flag.String("data", "", "The JSON string containing the ProcessingRequest data")
+)
```

Comment:
Could you move this group under the imports section (https://google.github.io/styleguide/go/decisions.html#flags).